autoPyTorch/datasets/tabular_dataset.py

```diff
@@ -73,7 +73,7 @@ def __init__(self,
                  seed: Optional[int] = 42,
                  train_transforms: Optional[torchvision.transforms.Compose] = None,
                  val_transforms: Optional[torchvision.transforms.Compose] = None,
-                 dataset_name: Optional[str] = None,
```

Comment:
Maybe `None` here is ok. My concern is that we where defining the name both in the dataset and in [here](https://github.com/automl/Auto-PyTorch/blob/988a3a23a356fd9492331792f25cf37935a921eb/autoPyTorch/datasets/base_dataset.py#L114)

Which one do you think makes more sense?